packages/ui/src/utils/nativeMaxAmount.ts

```diff
@@ -117,3 +122,69 @@ export const calculateBitcoinNativeFeeBuffer = async (): Promise<bigint> => {
     return 0n
   }
 }
+
+/**
+ * Calculates the fee buffer needed for a native SVM (e.g., Solana) or Eclipse transaction.
+ * Uses specific multipliers and default fees for each type.
+ * Falls back to a default fee if API data is unavailable or invalid.
+ *
+ * @param chainId - The chain ID of the SVM or Eclipse network.
+ * @returns The calculated fee buffer amount (in lamports for Solana, Wei for Eclipse) as a bigint, or a fallback buffer if estimation fails.
+ */
+export const calculateSvmNativeFeeBuffer = async (
+  chainId: number
+): Promise<bigint> => {
+  const isEclipseChain = chainId === 9286185
+  const multiplier = isEclipseChain
+    ? ECLIPSE_FEE_BUFFER_MULTIPLIER
+    : SOLANA_FEE_BUFFER_MULTIPLIER
+  const fallbackBuffer = isEclipseChain
+    ? ECLIPSE_DEFAULT_FEE_WEI * ECLIPSE_FEE_BUFFER_MULTIPLIER
+    : SOLANA_DEFAULT_FEE_LAMPORTS * SOLANA_FEE_BUFFER_MULTIPLIER
+
+  try {
+    const url = `${MAINNET_RELAY_API}/requests/v2?limit=20&sortBy=createdAt&originChainId=${chainId}`
+    const resp: any = await fetcher(url)
+    if (!resp || !Array.isArray(resp.requests) || resp.requests.length === 0) {
+      console.warn('No valid fees found in response. Using fallback buffer.')
+      return fallbackBuffer
+    }
+
+    // Extract fees from inTxs[0].fee
+    const fees: bigint[] = resp.requests
+      .map((r: any) => {
+        const feeStr = r?.data?.inTxs?.[0]?.fee
+        try {
+          return BigInt(feeStr)
+        } catch {
+          return null
+        }
+      })
+      .filter((f: bigint | null): f is bigint => f !== null)
+
+    if (fees.length === 0) {
+      return fallbackBuffer
+    }
+
+    // Sort fees in descending order
+    fees.sort((a, b) => (b > a ? 1 : b < a ? -1 : 0))
+
+    // Take the top 3 fees (or fewer if less than 3 are available)
+    const topFees = fees.slice(0, 3)
+
+    if (topFees.length === 0) {
+      return fallbackBuffer
+    }
+
+    // Sum the top fees
+    const sumOfTopFees = topFees.reduce((sum, current) => sum + current, 0n)
```

Comment:
Why are we summing the top fees? I thought we would find the average between them. That would make more sense right? Then multiply by the multiplier